src/benchmarkstt/metrics/core.py

```diff
@@ -159,6 +160,157 @@ def compare(self, ref: Schema, hyp: Schema) -> OpcodeCounts:
+                            abs(ne_hyp - ne_ref)
+        BEER (entity)   =   ----------------------
+                                ne_ref
+
+    - ne_hyp = number of detections of the entity in the hypothesis file
+    - ne_ref = number of detections of the entity in the reference file
+
+    The WA_BEER for a set of N entities is defined as the weighted average of the BEER for the set of
+    entities::
+
+        WA_BEER ([entity_1, ... entity_N) =  w_1*BEER (entity_1)*L_1/L + ... + w_N*BEER (entity_N))*L_N/L
+
+    which is equivalent to::
+
+                                            w_1*abs(ne_hyp_1 - ne_ref_1) + ... + w_N*abs(ne_hyp_N - ne_ref_N)
+        WA_BEER ([entity_1, ... entity_N) = ------------------------------------------------------------------
+                                                                            L
+
+    - L_1 =  number of occurrences of entity 1 in the reference document
+    - L = L_1 + ... + L_N
+
+    the weights being normalised by the tool:
+
+    - w_1 + ... + w_N = 1
+
+    The input file defines the list of entities and the weight per entity, w_n. It is processed as a json file with the
+    following structure::
+
+        { "entity_1":W_1, "entity_2" : W_2, "entity_3" :W_3 .. }
+
+    W_n being the non-normalized weight, the normalization of the weights is performed by the tool as::
+
+                    W_n
+        w_n =   ---------------
+                W_1 + ... +W_N
+
+    The minimum value for weight being 0.
+
+    """
+    def __init__(self, entities_file=''):
+        """
+        """
+        self._weight = []
+        self._entities = []
+
+        try:
+            with open(entities_file) as f:
+                data = json.load(f)
+            self._entities = list(data.keys())
+            weight = list(data.values())
+            self.set_weight(weight)
+        except IOError:
+            print('Input file Error')
+
+        return
+
+    def get_weight(self):
+        return self._weight
+
+    def set_weight(self, weight):
+        weight = [0 if w < 0 else w for w in weight]
+        sw = sum(weight)
+        if sw > 0:
+            self._weight = [w / sw for w in weight]
+
+    def get_entities(self):
+        return self._entities
+
+    def set_entities(self, entities):
+        self._entities = entities
+
+    # find the position of one entity
+    # an entity can contain more than one word
+    @staticmethod
+    def __find_pattern(search_list, complex_entity):
+        entity = ''.join(complex_entity).split(' ')
+        le = len(entity)
+        # complex_entity = [entity1 entity2 ...]
+        # the cursor sweep complex_entity to find consecutive entities entity1 ... entity2
+        cursor = 0
+        idx_found = []
+        for idx, elt in enumerate(search_list):
+            if elt == entity[cursor]:
+                cursor += 1
+                if cursor == le:
+                    idx_found.append([idx for idx in range(idx - le + 1, idx - le + 1 + le)])
```

Comment:
This could be simplified a little bit
```suggestion
                    idx_found.append(list(range(idx - le + 1, idx - le + 1 + le)))
```

Reply:
It can not be simplified this way. I renamed the idx of the upper level as idex_sl.

Reply:
OK. Now that the upper level has been named `idx_sl` I think that line can be changed to `idx_found.append(list(range(idx_sl - le + 1, idx_sl - le + 1 + le)))`. This just removes an extra variable definition (`idx`) that will make the code easier to understand.
I've run it with the new suggested line and all the tests still pass.

Reply:
done